integration/util.go

```diff
@@ -364,6 +366,77 @@ func (k *NSKubernetesClient) waitForDeploymentsToStabilizeWithTimeout(timeout ti
 	}
 }
 
+type getResult struct {
+	Items []Item `json:"items"`
+}
+
+type Item struct {
+	Metadata struct {
+		Name              string `json:"name"`
+		DeletionTimestamp string `json:"deletionTimestamp"`
+	} `json:"metadata"`
+}
+
+func (k *NSKubernetesClient) WaitForPodsDeletionWithTimeout(timeout time.Duration, interval time.Duration, podNames ...string) {
+	k.waitForResourceDeletionWithTimeout(timeout, interval, "Pod", podNames...)
+}
+
+func (k *NSKubernetesClient) WaitForDeploymentDeletionWithTimeout(timeout time.Duration, interval time.Duration, deployments ...string) {
+	k.waitForResourceDeletionWithTimeout(timeout, interval, "Deployment", deployments...)
+}
+
+func (k *NSKubernetesClient) waitForResourceDeletionWithTimeout(timeout time.Duration, interval time.Duration, kind string, resources ...string) {
```

Comment:
did you try something similar to [this](https://stackoverflow.com/a/71202736/4540147) using the event stream?

Reply:
verification for deletion is not necessary, hence removed this part.